File: .skills/openclaw-skills/skills/vnagin/zoom-manager-clawd/scripts/create_meeting.js
#!/usr/bin/env node
const fs = require('fs').promises;
const path = require('path');

// Load config
async function loadConfig() {
  const configPath = path.join(__dirname, '..', 'config.json');
  const data = await fs.readFile(configPath, 'utf8');
  return JSON.parse(data);
}

// Get OAuth access token
async function getAccessToken(config) {
  const url = 'https://api.zoom.us/oauth/token';
  const params = new URLSearchParams({
    grant_type: 'account_credentials',
    account_id: config.account_id
  }).toString();

  const response = await fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded',
      'Authorization': `Basic ${Buffer.from(`${config.client_id}:${config.client_secret}`).toString('base64')}`
    },
    body: params
  });

  if (!response.ok) {
    throw new Error(`Failed to get access token: ${response.status} ${response.statusText}`);
  }

  const data = await response.json();
  return data.access_token;
}

// Create a meeting
async function createMeeting(config, topic, startTime, durationMinutes) {
  const accessToken = await getAccessToken(config);
  const url = `https://api.zoom.us/v2/users/${config.user_id}/meetings`;
  const payload = {
    topic: topic,
    type: 2,
    start_time: startTime,
    duration: durationMinutes,
    timezone: 'UTC',
    settings: {
      host_video: true,
      participant_video: true,
      join_before_host: false,
      mute_upon_entry: false,
      approval_type: 0,
      audio: 'both',
      auto_recording: 'cloud'
    }
  };

  const response = await fetch(url, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${accessToken}`,
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(payload)
  });

  if (!response.ok) {
    throw new Error(`Failed to create meeting: ${response.status} ${response.statusText}`);
  }

  return await response.json();
}

// Main
async function main() {
  if (process.argv.length !== 5) {
    console.log('Usage: node create_meeting.js <topic> <start_time> <duration_minutes>');
    process.exit(1);
  }

  try {
    const config = await loadConfig();
    const topic = process.argv[2];
    const startTime = process.argv[3];
    const durationMinutes = parseInt(process.argv[4], 10);

    const meeting = await createMeeting(config, topic, startTime, durationMinutes);
    console.log('✅ Meeting created successfully!');
    console.log(`ID: ${meeting.id}`);
    console.log(`Join URL: ${meeting.join_url}`);
    console.log(`Start Time: ${meeting.start_time}`);
  } catch (error) {
    console.error('❌ Error:', error.message);
    process.exit(1);
  }
}

main();

File: .skills/openclaw-skills/skills/vnagin/zoom-manager-clawd/scripts/create_meeting.py
#!/usr/bin/env python3
import json
import requests
import sys
import os

# Load config
config_path = os.path.join(os.path.dirname(__file__), '..', 'config.json')
with open(config_path, 'r') as f:
    config = json.load(f)

ACCOUNT_ID = config['account_id']
CLIENT_ID = config['client_id']
CLIENT_SECRET = config['client_secret']
USER_ID = config['user_id']

def get_access_token():
    url = "https://api.zoom.us/oauth/token"
    data = {
        "grant_type": "account_credentials",
        "account_id": ACCOUNT_ID
    }
    response = requests.post(url, auth=(CLIENT_ID, CLIENT_SECRET), data=data)
    response.raise_for_status()
    return response.json()['access_token']

def create_meeting(topic, start_time, duration_minutes):
    access_token = get_access_token()
    url = f"https://api.zoom.us/v2/users/{USER_ID}/meetings"
    headers = {
        "Authorization": f"Bearer {access_token}",
        "Content-Type": "application/json"
    }
    payload = {
        "topic": topic,
        "type": 2,  # Scheduled meeting
        "start_time": start_time,
        "duration": duration_minutes,
        "timezone": "UTC",
        "settings": {
            "host_video": True,
            "participant_video": True,
            "join_before_host": False,
            "mute_upon_entry": False,
            "approval_type": 0,
            "audio": "both",
            "auto_recording": "cloud"
        }
    }
    response = requests.post(url, headers=headers, json=payload)
    response.raise_for_status()
    return response.json()

if __name__ == "__main__":
    if len(sys.argv) != 4:
        print("Usage: create_meeting.py <topic> <start_time> <duration_minutes>")
        sys.exit(1)
    
    topic = sys.argv[1]
    start_time = sys.argv[2]
    duration_minutes = int(sys.argv[3])
    
    try:
        meeting = create_meeting(topic, start_time, duration_minutes)
        print(f"Meeting created successfully!")
        print(f"ID: {meeting['id']}")
        print(f"Join URL: {meeting['join_url']}")
        print(f"Start Time: {meeting['start_time']}")
    except Exception as e:
        print(f"Error: {e}")